tests/Detector.Tests/Go/GoDetectorTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Detector.Go;
+using Microsoft.Oryx.Tests.Common;
+using Xunit;
+namespace Microsoft.Oryx.Detector.Tests.Go
+{
+    public class GoDetectorTest : IClassFixture<TestTempDirTestFixture>
+    {
+
+        private const string GoFileWithVersion = @"
+            module hello
+
+            go 1.16
+        ";
+
+        private const string GoFileWithNoVersions = @"
+            module hello
+        ";
+        private readonly string _tempDirRoot;
+
+        public GoDetectorTest(TestTempDirTestFixture testFixture)
+        {
+            _tempDirRoot = testFixture.RootDirPath;
+        }
+
+        // test empty source repo
+        [Fact]
+        public void Detect_ReturnsNull_WhenSourceDirectoryIsEmpty()
```

Comment:
rename: GoLang_Detector_ReturnsNull_WhenSourceDirectory_IsEmpty

Reply:
I do prefer not prefixing platform name, since the tests are under go directory and file. Additionally other platform tests do not prefix individual tests with platform name